pkg/influx/proxy.go

```
@@ -0,0 +1,124 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to create remotewrite.API: %w", err)
+	}
+
+	return newProxyWithClient(conf, client)
+}
+
+// newProxyWithClient creates the influx API server with the given config options and
+// the specified remotewrite client. It returns the HTTP server that is ready to Run.
+func newProxyWithClient(conf ProxyConfig, client remotewrite.Client) (*ProxyService, error) {
+	if conf.Logger == nil {
+		conf.Logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
+	}
+
+	recorder := NewRecorder(conf.Registerer)
+
+	var authMiddleware middleware.Interface
+	if conf.EnableAuth {
+		authMiddleware = middleware.NewHTTPAuth(conf.Logger)
+	} else {
+		authMiddleware = middleware.HTTPFakeAuth{}
+	}
+
+	server, err := server.NewServer(conf.Logger, conf.HTTPConfig, mux.NewRouter(), []middleware.Interface{authMiddleware})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create http server: %w", err)
+	}
+
+	api, err := NewAPI(conf.Logger, client, recorder)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create influx API: %w", err)
+	}
+
+	api.Register(server.Router)
+	err = recorder.RegisterVersionBuildTimestamp()
+	if err != nil {
+		return nil, fmt.Errorf("could not register version build timestamp: %w", err)
+	}
+
+	p := &ProxyService{
+		Logger:     conf.Logger,
+		config:     conf,
+		httpServer: server,
+	}
+	p.Service = services.NewBasicService(p.start, p.run, p.stop)
+	return p, nil
+}
+
+// Addr returns the net.Addr for the configured server. This is useful in case
+// it was started with port auto-selection so the port number can be retrieved.
+func (p *ProxyService) Addr() net.Addr {
+	return p.httpServer.Addr()
+}
+
+func (p *ProxyService) start(_ context.Context) error {
+	return nil
+}
+
+func (p *ProxyService) stop(_ error) error {
+	p.httpServer.Shutdown(nil)
+	return nil
+}
+
+func (p *ProxyService) run(servCtx context.Context) error {
+	go func() {
+		p.httpServer.Run()
```

this isn't quite right because we are swallowing the httpServer error return value

---

fixed